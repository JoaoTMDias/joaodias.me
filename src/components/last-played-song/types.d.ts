export interface IExternalServiceSongs {
	recenttracks: IRecenttracks;
}

export interface IRecenttracks {
	"@attr": IAttr;
	track: ITrack[];
}

export interface IAttr {
	page: string;
	perPage: string;
	user: string;
	total: string;
	totalPages: string;
}

export interface ITrack {
	artist: IAlbum;
	album: IAlbum;
	image: IImage[];
	streamable: string;
	date: IDateClass;
	url: string;
	name: string;
	mbid: string;
}

export interface IAlbum {
	mbid: string;
	"#text": string;
}

export interface IDateClass {
	uts: string;
	"#text": string;
}

export interface IImage {
	size: "extralarge" | "large" | "medium" | "small";
	"#text": string;
}
