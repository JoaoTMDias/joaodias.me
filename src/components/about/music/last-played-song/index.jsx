// Libraries
import React, { PureComponent } from 'react';
import styled from 'styled-components';

// Components
import asyncComponent from '../../../hoc/asyncComponent';

// Services
import MusicServices from '../../../../data/services/external-services';

const LastPlayedSongCard = asyncComponent(() => {
  return import(`./lastPlayedSongCard`);
});

/** ********
 ** Class: LastPlayedSong
 ** @type: Class
 ** @description: Retrieves info from Last.fm and displays on the widget
 ********* */
class LastPlayedSong extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      LastPlayedSong: [],
    };
  }

  componentDidMount() {
    this.fetchLatestSong();
  }

  componentWillUnmount() {
    this.state = {
      LastPlayedSong: [],
    };
  }

  fetchLatestSong() {
    MusicServices.getLatestSong()
      .then(response => {
        if (response.data) {
          this.setState({
            LastPlayedSong: response.data.recenttracks.track,
          });
        }
      })
      .catch(error => console.error('erro no lastfm: ', error));
  }

  renderLastPlayedSong() {
    const numberofSongs = 1;

    const LatestSong = this.state.LastPlayedSong.slice(0, numberofSongs).map(song => (
      <LastPlayedSongCard key={song.name} song={song} />
    ));

    return LatestSong;
  }

  render() {
    // console.log(this.state);

    return (
      <section id="last-played-song" className="l__container l__section">
        <div className="l__row">
          <Title tabIndex="0">Currently listening to...</Title>
          {this.renderLastPlayedSong()}
        </div>
      </section>
    );
  }
}

const Title = styled.h2`
  color: var(--color-gray8, ${props => props.theme.gray8});
  text-align: center;
  font-size: 1.125rem;
  font-family: var(--body-font-family, ${props =>props.theme.bodyFontFamily});
`;

export default LastPlayedSong;
