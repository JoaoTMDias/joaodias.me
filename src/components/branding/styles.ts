// Libraries
import { Link } from "gatsby";
import styled from "styled-components";

// Styling
export const Brand = styled(Link)`
	display: flex;
	align-self: flex-start;
	width: 3rem;
	text-decoration: none;
	margin: 0;
`;

export const Logo = styled.svg`
	display: inline-block;
	width: 3rem;
	height: 3rem;
	margin: 0;
	vertical-align: middle;
	transition-property: all;
	transition-duration: 112ms;
	transition-timing-function: var(--default-timing-function);

	.brand__hair {
		fill: var(--color-black, #030304);
	}

	.brand__glasses {
		fill: var(--color-black, #030304);
	}

	&:hover,
	&:focus {
		transform: scale(1.2) rotate(10deg);

		.brand__hair {
			fill: var(--color-primary);
		}

		.brand__glasses {
			transform: translateY(2px);
			fill: var(--color-primary);
		}
	}

	&:active {
		transform: scale(1.4) rotate(-10deg);

		.brand__hair {
			fill: var(--color-primary);
		}

		.brand__glasses {
			transform: translateY(2px);
			fill: var(--color-alternate);
		}
	}
`;
