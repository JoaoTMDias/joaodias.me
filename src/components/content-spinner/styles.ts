import styled, { keyframes, css } from "styled-components";
import { rem } from "polished";
import { IContentSpinnerProps } from "./types";

// /////////
// STYLING
// /////////

// Animations
const hideTimeout = keyframes`
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0;
  }
`;

const dashAnimation = keyframes`
  0% {
    stroke-dasharray: 1, 150;
    stroke-dashoffset: 0;
  }
  50% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -35;
  }
  100% {
    stroke-dasharray: 90, 150;
    stroke-dashoffset: -124;
  }
`;

const spinnerRotate = keyframes`
  100% {
    transform: rotate(360deg);
  }
`;

// Markup
export const ContentSpinnerWrapper = styled.div`
	width: 100%;
	height: ${({ fullPage, center, size }: IContentSpinnerProps) => {
		if (fullPage) {
			return "100vh";
		}
		if (center) {
			return "100%";
		}
		return `${size}px`;
	}};
	margin: 0 auto;
	position: relative;
	display: table;
	display: flex;
	flex-direction: row;
	justify-content: ${({ center }: IContentSpinnerProps) => (center ? "center" : "inherit")};
	align-items: ${({ center }: IContentSpinnerProps) => (center ? "center" : "inherit")};
	background-color: transparent;
	opacity: 1;
	user-select: none;
	pointer-events: none;

	${({ temporary, duration }: IContentSpinnerProps) => {
		if (temporary) {
			return css`
				animation-name: ${hideTimeout};
				animation-duration: 500ms;
				animation-delay: ${`${duration}ms`};
				animation-timing-function: var(--default-timing-function);
				animation-fill-mode: forwards;
			`;
		}
	}};

	.spinner {
		&__container {
			display: table-cell;
			vertical-align: middle;
			display: flex;
			z-index: 2;
			position: absolute;
			top: 50%;
			left: 50%;

			${({ size }: IContentSpinnerProps) => {
				if (size) {
					return css`
						margin-top: ${rem(`${-1 * (size * 0.5)}px`)};
						margin-right: 0;
						margin-bottom: 0;
						margin-left: ${rem(`${-1 * (size * 0.5)}px`)}

						width: ${rem(`${size}px`)};
						height: ${rem(`${size}px`)};
					`;
				}
			}};

			animation-name: ${spinnerRotate};
			animation-duration: ${({ duration }: IContentSpinnerProps) => `${duration}ms`};
			animation-timing-function: linear;
			animation-delay: 0s;
			animation-iteration-count: infinite;
			animation-direction: normal;
			animation-fill-mode: none;
			animation-play-state: running;
		}

		&__icon {
			stroke: ${({ color }: IContentSpinnerProps) => `${color}`};
			stroke-linecap: round;

			animation-name: ${dashAnimation};
			animation-duration: ${({ duration }: IContentSpinnerProps) => duration && `${duration * 0.5}ms`};
			animation-timing-function: var(--default-timing-function);
			animation-iteration-count: infinite;
			animation-delay: 0;
			animation-direction: normal;
			animation-fill-mode: none;
			animation-play-state: running;

			${({ delay }: IContentSpinnerProps) => {
				if (delay) {
					return css`
						animation-delay: ${delay && `${delay}ms`};
					`;
				}
			}};
		}
`;

export default ContentSpinnerWrapper;
