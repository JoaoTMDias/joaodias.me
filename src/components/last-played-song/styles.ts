import styled from "styled-components";
import { above } from "../../helpers/media-queries.helper";
import { rem } from "polished";
import { theme } from "../../helpers/theme.helper";

export const LastPlayedSongTitle = styled.h2`
	color: var(--color-gray8, #646b82);
	text-align: center;
	font-size: 1.125rem;
	font-family: var(--body-font-family);

	${theme.dark`
		color: var(--color-gray4);
	`};
`;

export const LastPlayedSong = styled.div`
	width: 100%;
	max-width: 30rem;
	min-width: 20rem;
	flex-direction: column;
	display: flex;
	margin: 0 auto;
	align-items: center;

	${above.large`
		max-width: 40rem;
		min-width: 35rem;
	`};
`;

export const InfoContainer = styled.div`
	display: flex;
	margin-left: 0;
	flex-direction: column;
	width: 100%;
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
	text-align: center;

	.track {
		font-family: var(--body-font-family);
		font-size: 1rem;
		font-weight: bold;
		margin-bottom: 0.75rem;
	}

	.track a {
		color: var(--color-gray9, #3a3e4c);
	}

	.album {
		font-size: 0.875rem;
		color: var(--color-gray8, #646b82);
	}

	.artist {
		font-size: 0.875rem;
	}

	.artist,
	.album {
		margin-bottom: 0.5rem;
	}

	.track,
	.track a,
	.url,
	.artist {
		white-space: nowrap;
		overflow: hidden;
		text-overflow: ellipsis;
	}

	${theme.dark`
		.track a,
		.artist {
			color: var(--color-gray3);
		}
		

		.album {
			color: var(--color-gray4);
		}
	`};
`;

export const RECORD_COVER_SIZE = 240;

export const Cover = styled.a`
	width: ${(props: IAlbumCoverProps) => {
		if (props.width) {
			return `${rem(`${props.width}px`)}`;
		}

		return `${rem(`${RECORD_COVER_SIZE}px`)}`;
	}};
	height: ${(props: IAlbumCoverProps) => {
		if (props.width) {
			return `${rem(`${props.width}px`)}`;
		}

		return `${rem(`${RECORD_COVER_SIZE}px`)}`;
	}};
	display: flex;
	justify-content: center;
	align-items: center;
	margin-bottom: var(--global-margin);

	box-shadow: 0px 10px 30px -5px rgba(0, 0, 0, 0.3);
	transition: box-shadow 500ms;
	will-change: transform;

	&:hover {
		box-shadow: 0px 30px 100px -10px rgba(0, 0, 0, 0.4);
	}

	img {
		position: absolute;
	}

	${above.large`
		justify-content: flex-start;
	`};
`;

export const Album = styled.img`
	width: ${(props: IAlbumCoverProps) => {
		if (props.width) {
			return `${rem(`${props.width}px`)}`;
		}

		return `${rem(`${RECORD_COVER_SIZE}px`)}`;
	}};
	height: ${(props: IAlbumCoverProps) => {
		if (props.width) {
			return `${rem(`${props.width}px`)}`;
		}

		return `${rem(`${RECORD_COVER_SIZE}px`)}`;
	}};
	z-index: 1;
	box-shadow: 0px 2px 10px rgba(0, 0, 0, 0.2);
	border-radius: ${rem("8px")};
	transition: transform 250ms;
	background-color: var(--color-gray7, #82889b);
	color: var(--color-white, #ffffff);
`;

export interface IAlbumCoverProps {
	url: string;
	name: string;
	artist: string;
	src: string;
	alt: string;
	width?: number;
}
