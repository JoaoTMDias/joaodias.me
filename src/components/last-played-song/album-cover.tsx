// Libraries
import * as React from "react";
import { useSpring, animated } from "react-spring";
import { IAlbumCoverProps, Cover, Album } from "./styles";

const calc = (x: number, y: number) => [-(y - window.innerHeight / 2) / 10, (x - window.innerWidth / 2) / 10, 1.1];
const trans = (x: number, y: number, s: number) => `perspective(600px) rotateX(${x}deg) rotateY(${y}deg) scale(${s})`;

export const defaultProps = {
	width: 240,
};

/**
 * @description Recent Song Album Cover
 * @author Jo√£o Dias
 * @date 2019-02-16
 * @returns {React.FunctionComponent<IAlbumCoverProps>}
 */
const AlbumCover: React.FunctionComponent<IAlbumCoverProps> = props => {
	const { src, alt, url, name, artist, width } = props;
	const { ...rest } = props;

	const [animation, setAnimation] = useSpring(() => ({
		xys: [0, 0, 1],
		config: { mass: 5, tension: 350, friction: 40 },
	}));

	const AnimatedCover = animated(Cover);

	const title = `Click/Enter to listen to ${name} by ${artist} on Last.fm`;

	return (
		<AnimatedCover
			data-testid="component-animated-cover"
			href={url}
			title={title}
			onMouseMove={
				(event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) =>
					setAnimation({ xys: calc(event.clientX, event.clientY) })
				// eslint-disable-next-line react/jsx-curly-newline
			}
			onMouseLeave={() => setAnimation({ xys: [0, 0, 1] })}
			style={{ transform: animation.xys.interpolate(trans) }}
			{...rest}
		>
			<Album {...rest} src={src} width={width} height={width} alt={alt} />
		</AnimatedCover>
	);
};

AlbumCover.defaultProps = defaultProps;

export default React.memo(AlbumCover);
