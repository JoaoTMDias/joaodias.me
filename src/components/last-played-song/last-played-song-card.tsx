// Libraries
import React from "react";
import AlbumCover from "./album-cover";
import { ITrack } from "./types";
import { LastPlayedSong, InfoContainer } from "./styles";

export interface ILastPlayedSongCardProps {
	song: ITrack;
}

/**
 * Last Played Song Card
 *
 * @export
 * @param {ILastPlayedSongCardProps} props
 * @returns
 */
export const LastPlayedSongCard: React.FunctionComponent<ILastPlayedSongCardProps> = ({ song }) => {
	return (
		<LastPlayedSong>
			<AlbumCover
				url={song.url}
				name={song.name}
				artist={song.artist["#text"]}
				src={song.image[2]["#text"]}
				alt={`Album Cover for ${song.album["#text"]}`}
			/>

			<InfoContainer>
				<h2 className="track">
					<a
						href={song.url}
						data-testid="last-played-title"
						aria-label={`Song Name: ${song.name}`}
						title={`${song.name}`}
					>
						{song.name}
					</a>
				</h2>
				<p className="artist" data-testid="last-played-artist" aria-label={`Artist/Band Name: ${song.artist["#text"]}`}>
					{song.artist["#text"]}
				</p>
				<span className="album" data-testid="last-played-album" aria-label={`Album Name: ${song.album["#text"]}`}>
					{song.album["#text"]}
				</span>
			</InfoContainer>
		</LastPlayedSong>
	);
};

export default LastPlayedSongCard;
