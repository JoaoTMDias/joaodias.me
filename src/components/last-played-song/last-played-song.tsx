// Libraries
import React from "react";

// Services
import { LastPlayedSongCard } from "./last-played-song-card";
import { LastPlayedSongTitle } from "./styles";
import { ITrack } from "./types";
import { LAST_FM_URL } from "../../data/services/config";
import { useAsyncHook } from "../../helpers";

/**
 * Retrieves info from Last.fm and displays on the widget
 *
 * @export
 * @returns {JSX.Element[] | null}
 */
export const LastPlayedSong = () => {
	const [result, loading] = useAsyncHook(LAST_FM_URL);

	/**
	 * Renders the last played song
	 *
	 * @returns
	 */
	function renderLastPlayedSong() {
		if (result) {
			const song = result as ITrack;
			return <LastPlayedSongCard key={song.name} song={song} />;
		}

		return null;
	}

	if (loading) {
		return null;
	}

	return (
		<section id="last-played-song" className="layout__container layout__section">
			<div className="layout__row">
				<LastPlayedSongTitle>Currently listening to...</LastPlayedSongTitle>
				{renderLastPlayedSong()}
			</div>
		</section>
	);
};

export default LastPlayedSong;
