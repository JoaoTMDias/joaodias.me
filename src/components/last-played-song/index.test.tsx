import React from "react";
import { render, cleanup } from "@testing-library/react";
import AlbumCover, { defaultProps } from "./album-cover";
import { LastPlayedSong } from "./last-played-song";
import { LastPlayedSongCard, ILastPlayedSongCardProps } from "./last-played-song-card";
import { IAlbumCoverProps } from "./styles";

afterEach(cleanup);

describe("<LastPlayedSong />", () => {
	it("should render without errors", () => {
		const component = render(<LastPlayedSong />);

		expect(component).toMatchSnapshot();
	});
});

describe("<LastPlayedSongCard />", () => {
	const props: ILastPlayedSongCardProps = {
		song: {
			artist: {
				mbid: "83b9cbe7-9857-49e2-ab8e-b57b01038103",
				"#text": "Pearl Jam",
			},
			album: {
				mbid: "",
				"#text": "Vault 9: Live in Seattle 12/8/93",
			},
			image: [
				{
					size: "small",
					"#text": "https://lastfm.freetls.fastly.net/i/u/34s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg",
				},
				{
					size: "medium",
					"#text": "https://lastfm.freetls.fastly.net/i/u/64s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg",
				},
				{
					size: "large",
					"#text": "https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg",
				},
				{
					size: "extralarge",
					"#text": "https://lastfm.freetls.fastly.net/i/u/300x300/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg",
				},
			],
			streamable: "0",
			date: {
				uts: "1577995462",
				"#text": "02 Jan 2020, 20:04",
			},
			url: "https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)",
			name: "Animal (Live)",
			mbid: "4c43b91a-5aba-4f1e-ac14-27f42df5c04d",
		},
	};

	it("should render without errors", () => {
		const component = render(<LastPlayedSongCard {...props} />);

		expect(component).toMatchSnapshot();
	});
});

describe("<AlbumCover />", () => {
	const props: IAlbumCoverProps = {
		...defaultProps,
		url: "https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)",
		artist: "Pearl Jam",
		src: "https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg",
		name: "Animal (Live)",
		alt: "Album Cover for Vault 9: Live in Seattle 12/8/1993",
		width: defaultProps.width,
	};

	it("should render without errors", () => {
		const component = render(<AlbumCover {...props} />);

		expect(component).toMatchSnapshot();
	});
});
