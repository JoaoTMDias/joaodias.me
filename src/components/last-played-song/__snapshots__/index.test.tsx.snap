// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumCover /> should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        alt="Album Cover for Vault 9: Live in Seattle 12/8/1993"
        class="sc-bxivhb fgdRAd"
        data-testid="component-animated-cover"
        href="https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)"
        name="Animal (Live)"
        src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
        title="Click/Enter to listen to Animal (Live) by Pearl Jam on Last.fm"
        width="240"
      >
        <img
          alt="Album Cover for Vault 9: Live in Seattle 12/8/1993"
          class="sc-ifAKCX eUovE"
          height="240"
          name="Animal (Live)"
          src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
          width="240"
        />
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      alt="Album Cover for Vault 9: Live in Seattle 12/8/1993"
      class="sc-bxivhb fgdRAd"
      data-testid="component-animated-cover"
      href="https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)"
      name="Animal (Live)"
      src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
      style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
      title="Click/Enter to listen to Animal (Live) by Pearl Jam on Last.fm"
      width="240"
    >
      <img
        alt="Album Cover for Vault 9: Live in Seattle 12/8/1993"
        class="sc-ifAKCX eUovE"
        height="240"
        name="Animal (Live)"
        src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
        width="240"
      />
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LastPlayedSong /> should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LastPlayedSongCard /> should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bwzfXH klsUJK"
      >
        <a
          alt="Album Cover for Vault 9: Live in Seattle 12/8/93"
          class="sc-bxivhb fgdRAd"
          data-testid="component-animated-cover"
          href="https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)"
          name="Animal (Live)"
          src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
          style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
          title="Click/Enter to listen to Animal (Live) by Pearl Jam on Last.fm"
          width="240"
        >
          <img
            alt="Album Cover for Vault 9: Live in Seattle 12/8/93"
            class="sc-ifAKCX eUovE"
            height="240"
            name="Animal (Live)"
            src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
            width="240"
          />
        </a>
        <div
          class="sc-htpNat bSzeMz"
        >
          <h2
            class="track"
          >
            <a
              aria-label="Song Name: Animal (Live)"
              href="https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)"
              title="Animal (Live)"
            >
              Animal (Live)
            </a>
          </h2>
          <p
            aria-label="Artist/Band Name: Pearl Jam"
            class="artist"
          >
            Pearl Jam
          </p>
          <span
            aria-label="Album Name: Vault 9: Live in Seattle 12/8/93"
            class="album"
          >
            Vault 9: Live in Seattle 12/8/93
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bwzfXH klsUJK"
    >
      <a
        alt="Album Cover for Vault 9: Live in Seattle 12/8/93"
        class="sc-bxivhb fgdRAd"
        data-testid="component-animated-cover"
        href="https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)"
        name="Animal (Live)"
        src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
        style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
        title="Click/Enter to listen to Animal (Live) by Pearl Jam on Last.fm"
        width="240"
      >
        <img
          alt="Album Cover for Vault 9: Live in Seattle 12/8/93"
          class="sc-ifAKCX eUovE"
          height="240"
          name="Animal (Live)"
          src="https://lastfm.freetls.fastly.net/i/u/174s/c6aaab6608ee5ed79cac652a7a3d3cd9.jpg"
          width="240"
        />
      </a>
      <div
        class="sc-htpNat bSzeMz"
      >
        <h2
          class="track"
        >
          <a
            aria-label="Song Name: Animal (Live)"
            href="https://www.last.fm/music/Pearl+Jam/_/Animal+(Live)"
            title="Animal (Live)"
          >
            Animal (Live)
          </a>
        </h2>
        <p
          aria-label="Artist/Band Name: Pearl Jam"
          class="artist"
        >
          Pearl Jam
        </p>
        <span
          aria-label="Album Name: Vault 9: Live in Seattle 12/8/93"
          class="album"
        >
          Vault 9: Live in Seattle 12/8/93
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
